libamqpprox/amqpprox_maybesecuresocketadaptor.h

```diff
@@ -331,6 +371,46 @@ class MaybeSecureSocketAdaptor {
                                                              handler);
         }
 
+        if (!d_alarmed && d_dataRateAlarm.remainingQuota() == 0) {
+            if (d_dataRateTimerStarted) {
+                LOG_INFO << "Data Rate Alarm: Hit "
+                         << d_dataRateAlarm.getQuota() << " bytes/s";
+
+                // TODO: Better limit alarm debouncing?
+                d_alarmed = true;
+            }
+            else {
+                // We have hit our quota but we haven't started the refresh
+                // timer yet so it's probable the usage has never reset. Start
+                // the refresh timer for this connection and continue until we
+                // hit it a second time.
+
+                this->onTimer(boost::system::error_code());
+            }
+        }
+
+        if (d_dataRateLimit.remainingQuota() == 0) {
+            if (d_dataRateTimerStarted) {
+                d_dataRateTimer.cancel();
+                d_dataRateTimer.expires_at(d_dataRateTimer.expiry());
+                d_dataRateTimer.async_wait(
```

Comment:
It would be easier to understand the code in future, if there is some comment about both two different expires time for `d_dataRateTimer`?